src/main/java/teammates/common/datatransfer/questions/FeedbackNumericalScaleQuestionDetails.java

```diff
@@ -488,29 +456,101 @@ public String getQuestionResultStatisticsCsv(
-                           + SanitizationHelper.sanitizeForCsv(isRecipientGeneral
-                                                      ? "General"
-                                                      : bundle.getNameForEmail(recipient))
-                           + ','
-                           + df.format(average.get(recipient)) + ','
-                           + df.format(min.get(recipient)) + ','
-                           + df.format(max.get(recipient))
-                           + (showAvgExcludingSelf ? ',' + averageScoreExcludingSelfText : "")
-                           + Const.EOL);
+            updateQuestionResultStatisticsCsv(csvBody, bundle, recipient, average,
+                    min, max, showAvgExcludingSelf, averageExcludingSelf);
         }
 
         return csvHeader + csvBody.toString();
     }
 
+    /** adds a single record to the QuestionResultStatisticsCsv.
+     * @return void
+     */
+    private void updateQuestionResultStatisticsCsv(StringBuilder csvBody, FeedbackSessionResultsBundle bundle,
+            String recipient, Map<String, Double> average, Map<String, Double> min,
+            Map<String, Double> max, boolean showAvgExcludingSelf, Map<String, Double> averageExcludingSelf) {
+
+        String recipientTeam = bundle.getTeamNameForEmail(recipient);
+        boolean isRecipientGeneral = recipient.equals(Const.GENERAL_QUESTION);
+
+        DecimalFormat df = customDecimalFormat(0, 5, "down");
+
+        Double averageScoreExcludingSelf = averageExcludingSelf.get(recipient);
+        String averageScoreExcludingSelfText =
+                getAverageExcludingSelfText(showAvgExcludingSelf, df, averageScoreExcludingSelf);
+
+        csvBody.append(SanitizationHelper.sanitizeForCsv(recipientTeam) + ','
+                + SanitizationHelper.sanitizeForCsv(isRecipientGeneral
+                    ? "General"
+                    : bundle.getNameForEmail(recipient))
+                + ','
+                + df.format(average.get(recipient)) + ','
+                + df.format(min.get(recipient)) + ','
+                + df.format(max.get(recipient))
+                + (showAvgExcludingSelf ? ',' + averageScoreExcludingSelfText : "")
+                + Const.EOL);
+    }
+
+    /** creates a DecimalFormat with a given minimum & maximum fraction digits, and a rounding mode.
+     * @return DecimalFormat
+     */
+    private DecimalFormat customDecimalFormat(int min, int max, String roundingMode) {
+        DecimalFormat df = new DecimalFormat();
+        df.setMinimumFractionDigits(min);
+        df.setMaximumFractionDigits(max);
+
+        if ("down".equals(roundingMode)) {
+            df.setRoundingMode(RoundingMode.DOWN);
+        } else if ("up".equals(roundingMode)) {
+            df.setRoundingMode(RoundingMode.UP);
+        }
+        return df;
+    }
+
+    private String fragmentHtmlUpdate(Set<String> recipientSet, List<String> hiddenRecipients,
```

Comment:
I have the same comments as `updateQuestionResultStatisticsCsv`, but even more applicable here since the number of method parameters is very large, beyond reasonable.